erpnext/stock/doctype/delivery_trip/delivery_trip.py

```diff
@@ -45,137 +54,248 @@ def update_delivery_notes(self, delete=False):
 		delivery_notes = [get_link_to_form("Delivery Note", note) for note in delivery_notes]
 		frappe.msgprint(_("Delivery Notes {0} updated".format(", ".join(delivery_notes))))
 
+	def process_route(self, optimize):
+		"""
+		Estimate the arrival times for each stop in the Delivery Trip.
+		If `optimize` is True, the stops will be re-arranged, based
+		on the optimized order, before estimating the arrival times.
+
+		Args:
+			optimize (bool): True if route needs to be optimized, else False
+		"""
+
```

Comment:
This should work only in the case if Google Maps is enabled

Reply:
Added a check.